src/desktop/bundled/kis_tablet/kis_tablet_support_win.cpp

```diff
@@ -141,23 +141,37 @@ static inline bool isMouseEventType(QEvent::Type t)
 
 QWindowsWinTab32DLL QWindowsTabletSupport::m_winTab32DLL;
 
-void KisTabletSupportWin::init()
+static QMetaObject::Connection conn;
+
+bool KisTabletSupportWin::init()
 {
     if (!QWindowsTabletSupport::m_winTab32DLL.init()) {
         qWarning() << "Failed to initialize Wintab";
-        return;
+        return false;
     }
 
+    quit();
     QTAB = QWindowsTabletSupport::create();
+    if(!QTAB) {
+        return false;
+    }
 
     // Refresh tablet context after tablet rotated, screen added, etc.
-    QObject::connect(qApp->primaryScreen(), &QScreen::geometryChanged,
+    conn = QObject::connect(qApp->primaryScreen(), &QScreen::geometryChanged,
                      [=](const QRect & ){
                          delete QTAB;
                          QTAB = QWindowsTabletSupport::create();
                      });
 }
 
+void KisTabletSupportWin::quit()
+{
+    QObject::disconnect(conn);
+    conn = QMetaObject::Connection{};
+    delete QTAB;
```

Comment:
`std::unique_ptr`?

Reply:
Used that here and also in similar situations in tabletinput.cpp.